plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReloader.java

```diff
@@ -0,0 +1,331 @@
+        }
+
+        String localNodeId = clusterService.localNode().getId();
+        // auto reload all models of this local ml node
+        threadPool.generic().submit(() -> {
+            try {
+                autoReloadModelByNodeId(localNodeId);
+            } catch (ExecutionException | InterruptedException e) {
+                log.error("the model auto-reloading has exception,and the message is: {}", ExceptionUtils.getStackTrace(e));
+                throw new MLException(e);
+            }
+        });
+    }
+
+    /**
+     * auto reload all the models under the node id<br> the node must be a ml node<br>
+     *
+     * @param localNodeId node id
+     */
+    @VisibleForTesting
+    void autoReloadModelByNodeId(String localNodeId) throws ExecutionException, InterruptedException {
+        StepListener<SearchResponse> queryTaskStep = new StepListener<>();
+        StepListener<SearchResponse> getRetryTimesStep = new StepListener<>();
+        StepListener<IndexResponse> saveLatestRetryTimesStep = new StepListener<>();
+
+        if (!clusterService.state().metadata().indices().containsKey(ML_TASK_INDEX)) {
+            // ML_TASK_INDEX did not exist,do nothing
+            return;
+        }
+
+        queryTask(localNodeId, ActionListener.wrap(queryTaskStep::onResponse, queryTaskStep::onFailure));
+
+        getRetryTimes(localNodeId, ActionListener.wrap(getRetryTimesStep::onResponse, getRetryTimesStep::onFailure));
+
+        queryTaskStep.whenComplete(searchResponse -> {
+            SearchHit[] hits = searchResponse.getHits().getHits();
+            if (CollectionUtils.isEmpty(hits)) {
+                return;
+            }
+
+            getRetryTimesStep.whenComplete(getReTryTimesResponse -> {
+                int retryTimes = 0;
+                // if getReTryTimesResponse is null,it means we get retryTimes at the first time,and the index
+                // .plugins-ml-model-reload doesn't exist,so we should let retryTimes be zero(init value)
+                // we don't do anything
+                // if getReTryTimesResponse is not null,it means we have saved the value of retryTimes into the index
+                // .plugins-ml-model-reload,so we get the value of the field MODEL_LOAD_RETRY_TIMES_FIELD
+                if (getReTryTimesResponse != null) {
+                    Map<String, Object> sourceAsMap = getReTryTimesResponse.getHits().getHits()[0].getSourceAsMap();
+                    retryTimes = (Integer) sourceAsMap.get(MODEL_LOAD_RETRY_TIMES_FIELD);
+                }
+
+                // According to the node id to get retry times, if more than the max retry times, don't need to retry
+                // that the number of unsuccessful reload has reached the maximum number of times, do not need to reload
+                if (retryTimes > autoReloadMaxRetryTimes) {
+                    log.info("Node: {} has reached to the max retry limit, failed to load models", localNodeId);
+                    return;
```

Comment:
Before return, should we check how long the node has been in the max retry status and reset to 0 after a substantial time? It looks to me the node will never reload forever once reached maximum retry times.

Reply:
in the first comment of yours, I found ml_task index and ml_model index are both definition in [ml_task index](https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/indices/MLIndex.java), so if I add 2 new fields in ml_task,the ml_model index will have these 2 fields,too. It sounds that give ml_model index redundant attributes.
in the second comment of yours, when we discussed the design earlier, if the maximum retry times is reached, instead of automatically reloading, the model need to be loaded manually.

Reply:
Cool. Let's keep this logic then. But we should try to define a new type of ML Task for auto reload, and reuse MLTask  to store the max_retry field, etc.

Reply:
thanks~